docs/code-quality-tasks/module-graph-import-issues-report.md

```diff
@@ -0,0 +1,191 @@
+# Module Graph and Import Issues Report
+
+Based on the analysis of the codebase against AGENTS.md guidelines, the following module graph and import issues were found:
+
+## 1. Circular Dependencies Between Modules
+
+### Critical Circular Dependency: `@commontools/builder` ↔ `@commontools/runner`
+
+**Builder → Runner imports:**
+- `packages/builder/src/built-in.ts` imports from `@commontools/runner`
+- `packages/builder/src/schema-to-ts.ts` imports from `@commontools/runner`
+- `packages/builder/src/utils.ts` imports from `@commontools/runner`
+
+**Runner → Builder imports:**
+- `packages/runner/src/cfc.ts` imports from `@commontools/builder`
+- `packages/runner/src/recipe-manager.ts` imports from `@commontools/builder`
+- `packages/runner/src/runner.ts` imports from `@commontools/builder`
+- `packages/runner/src/env.ts` imports from `@commontools/builder`
+- `packages/runner/src/doc-map.ts` imports from `@commontools/builder`
+- `packages/runner/src/cell.ts` imports from `@commontools/builder`
+
+**Impact**: This circular dependency makes it impossible to:
+- Test these modules in isolation
+- Understand the module hierarchy
+- Reuse either module independently
+- Avoid loading unnecessary code
+
+**Recommendation**: Extract shared types and interfaces to a separate package like `@commontools/core-types` or `@commontools/shared`.
+
+## 2. Default Exports Instead of Named Exports
+
+The following files violate the "prefer named exports over default exports" guideline:
+
+### Component Files
+- `packages/charm/src/iframe/recipe.ts`
+- `packages/charm/src/commands.ts`
+- `packages/html/src/render.ts`
+- `packages/html/src/path.ts`
+- `packages/memory/clock.ts` - `export default new Clock();`
+
+### Plugin Files
+- `packages/deno-vite-plugin/src/index.ts`
+- `packages/deno-vite-plugin/src/resolvePlugin.ts`
+- `packages/deno-vite-plugin/src/prefixPlugin.ts`
+
+### View Components
+- `packages/jumble/src/assets/ShapeLogo.tsx`
+- `packages/jumble/src/views/CharmView.tsx`
+- `packages/jumble/src/views/ErrorBoundaryView.tsx`
+- `packages/jumble/src/views/DebugView.tsx`
+- Multiple other view files in `packages/jumble/src/views/`
+
+**Example Fix**:
+```typescript
+// Bad
+export default class Clock { ... }
+
+// Good
+export class Clock { ... }
+
+// Bad
+export default new Clock();
+
+// Good
+export const clock = new Clock();
+// Or better: export a factory function
+export function createClock(): Clock { ... }
+```
+
+## 3. Missing or Incorrect Import Grouping
```

Comment:
it'd be nice to use consistent ordering here, but doesn't seem like lint/fmt can ensure that -- this tenant could be lower priority than others as a lint preference rather than a functional change